src/utils.js

```diff
@@ -76,43 +82,49 @@ function extractBasis(sceneData, cell) {
     });
 }
 
-function validateSceneData(sceneData) {
-    // TODO
-}
-
-export function ThreeDSceneDataToMaterial(sceneData) {
-    validateSceneData(sceneData);
-    const lattice = extractLattice(sceneData);
-    const basis = extractBasis(sceneData, lattice.vectorArrays);
+/**
+ * Converts a given scene data to a material.
+ * Lattice is constructed from the LineSegments object.
+ * Basis is constructed based on all SphereMesh objects.
+ */
+export function ThreeDSceneDataToMaterial(scene) {
+    const lattice = extractLattice(scene);
+    const basis = extractBasis(scene, lattice.vectorArrays);
     basis.toCrystal();
     return new Made.Material({
-        name: sceneData.object.children[1].name,
+        name: scene.getObjectByProperty("type", "Group").name,
         lattice: lattice.toJSON(),
         basis: basis.toJSON(),
     });
 }
 
+/**
+ * Converts given materials to scene data.
+ * The first material is used as parent and it's unit cell is used in case multiple materials are passed.
+ * Other materials are added as a group under the first material with their cell hidden by default.
+ * The sites are slightly shifted along X axis if multiple materials are passed.
+ */
 export function materialsToThreeDSceneData(materials) {
     const wave = new Wave({
-        DOMElement: document.createElement("div"),
         structure: materials[0],
-        cell: materials[0].Lattice.unitCell
+        cell: materials[0].Lattice.unitCell,
+        DOMElement: document.createElement("div")
     });
-    wave.scene.remove(wave.camera);
-
-    const lightsGroup = new THREE.Group();
-    lightsGroup.name = "Lights - DO NOT MODIFY";
-    const directionalLight = new THREE.DirectionalLight("#FFFFFF");
-    directionalLight.name = "Directional - DO NOT MODIFY";
-    const ambientLight = new THREE.AmbientLight("#202020");
-    ambientLight.name = "Ambient - DO NOT MODIFY";
-    directionalLight.position.copy(new THREE.Vector3(0.2, 0.2, -1).normalize());
-    directionalLight.intensity = 1.2;
-    lightsGroup.add(directionalLight);
-    lightsGroup.add(ambientLight);
-    wave.scene.add(lightsGroup);
-    lightsGroup.position.set(-50, 0, 10);
-    wave.render();
-
+    if (materials.length > 1) {
+        wave.structureGroup.name = "New Material";
+        materials.slice(1).forEach(material => {
+            material.toCartesian();
+            const structureGroup = new THREE.Group();
+            structureGroup.name = material.name || material.formula;
+            const sitesGroup = wave.createSitesGroup(material.Basis);
+            structureGroup.add(sitesGroup);
+            const unitCellObject = wave.createUnitCellObject(material.Lattice.unitCell);
+            unitCellObject.visible = false;
+            structureGroup.add(unitCellObject);
+            structureGroup.position.set(2, 0, 0); //slightly shift along x axis
```

Comment:
I'd make the shift one of parameters of the function with default value of 2, 0, 0

Reply:
Done.